File: client/src/app/estimate/models/_2d.ts
import { _1d } from "./_1d";

export class _2d {
    public area!: number;
    public _1d_arr: [_1d] = [new _1d()];
    public total_plywood_sheets!: number;
    public plywood_sheet_wastage!: number;
    public dimension = "ft";
    public material!: string;
    public name!: string;
    public area_label!: string;
    public converted_label!: string;
    public conversion_dimension = "ft";

    constructor() { }

}

File: server/src/main/java/com/airtajal/model/_2d.java
package com.airtajal.model;

import lombok.AccessLevel;
import lombok.Data;
import lombok.experimental.FieldDefaults;
import java.util.Set;

import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import java.io.Serializable;

@Data
@FieldDefaults(level = AccessLevel.PRIVATE)
@Entity
public class _2d implements Serializable {
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    Long id;

    String name, label, dimension, material, conversion_dimension;
    Integer x, y, thikness, area, total_plywood_sheets, plywood_sheet_wastage;
    Set<_1d> _1d_arr;
}
